### PR TITLE
Automate string management

### PR DESCRIPTION
**Pull request by anuragajwani**
# Problem
String management on the JS SDK is hard. When updating string we have to copy and paste from our online source of truth into the SDK. This process involves designers updating strings and then developers copy and pasting them into the SDK in multiple languages.

# Solution
Use bisu to fetch the spreadsheet updated by designers to automatically update strings in the SDK (only for English now as Spanish needs updating in the spreadsheet)

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [ ] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [ ] Have new automated tests been implemented?
- [ ] Have new manual tests been written down?
- [ ] Have any new strings been translated?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-812-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-812-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://812-pr-onfido-sdk-ui-onfido.surge.sh

File: src/locales/en.json [hunk 1]
```diff
@@ -247,23 +247,14 @@
...
-  },
   "webcam_permissions": {
     "allow_access": "Allow camera access",
     "enable_webcam_for_selfie": "When prompted, you must enable camera access to continue",
     "click_allow": "We cannot verify you without using your camera",
     "allow": "Allow",
     "why": "Why do I need to do this?",
     "if_denied": "If you deny camera access, you won't be able to take pictures and complete verification process.",
-    "enable_webcam": "Enable camera",
+    "enable_webcam": "Enable webcam",
```
Comment (stefaniacardenas) [inline src/locales/en.json]: I think this string is shared between desktop and mobile. IMO, "Enable webcam" doesn't sound good on mobile.

File: src/locales/en.json [hunk 2]
```diff
@@ -316,41 +307,41 @@
...
       "instruction": "Please try again"
     },
     "glare_detected": {
       "message": "Glare detected",
       "instruction": "All details should be clear and readable"
     },
     "sms_failed": {
-      "message": "Something's gone wrong",
-      "instruction": "Copy the below link to your mobile instead"
+      "message": "Copy the link to your phone",
```
Comment (stefaniacardenas) [inline src/locales/en.json]: All our errors follow the same structure,`message` that informs the user about what happened and `instruction` which tells the user what to do or what they did wrong. Why are we changing the format here? @raemarielee

File: src/locales/en.json [hunk 3]
```diff
@@ -316,41 +307,41 @@
...
+      "instruction": "Something’s gone wrong"
     },
     "sms_overuse": {
-      "message": "Too many resend attempts",
-      "instruction": "Copy the below link to your mobile instead"
+      "message": "Copy the link to your phone",
+      "instruction": "Too many failed attempts"
```
Comment (stefaniacardenas) [inline src/locales/en.json]: Same here, why are we switching the instruction and the message around?

File: src/locales/en.json [hunk 4]
```diff
@@ -316,41 +307,41 @@
...
     "lazy_loading": {
       "message": "An error occurred while loading the component"
     },
     "invalid_number": {
-      "message": "Check your mobile number is correct"
+      "message": "Check that your number is correct"
     },
     "generic_client_error": {
-      "message": "Something's gone wrong",
+      "message": "Something’s gone wrong",
```
Comment (stefaniacardenas) [inline src/locales/en.json]: Why using a different character for the apostrophe? Are they equivalent?

File: src/locales/en.json [hunk 5]
```diff
@@ -316,41 +307,41 @@
...
     },
     "camera_not_working": {
-      "message": "Your camera isn’t working",
-      "instruction": "It may be disconnected or not functional. <fallback>Use your mobile</fallback> to continue verification."
+      "message": "Camera not working",
+      "instruction": "It may be disconnected. <fallback>Try using your phone instead</fallback>."
     },
     "camera_not_working_no_fallback": {
-      "instruction": "Make sure your device has a working camera"
+      "instruction": "Make sure your device's camera works"
```
Comment (stefaniacardenas) [inline src/locales/en.json]: Why this apostrophe is different from the one on line 332?

File: translatable.yml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+type: RoR
+
+dictionary:
+ type: google_sheet
+ sheet_id: 1B9RnT3vscNM8YGtIpzqNlSPCQUdHrcJ3cGPSGOtHmoA
```
Comment (stefaniacardenas) [inline translatable.yml]: Is it ok to have this in a public repo?

File: src/locales/en.json [hunk 7]
```diff
@@ -172,7 +172,7 @@
   },
```
Comment (stefaniacardenas) [inline src/locales/en.json]: Has this file been updated automatically by bisu?

File: src/locales/en.json (hunk 1 above)
Comment (anuragajwani) [reply src/locales/en.json]: sorry didn't change it on the spreadsheet, should I ? @raemarielee

File: src/locales/en.json (hunk 4 above)
Comment (anuragajwani) [reply src/locales/en.json]: no idea, I can change the spreadsheet if need be

File: translatable.yml (hunk 6 above)
Comment (anuragajwani) [reply translatable.yml]: they shouldn't be able to edit it, it's only for download

File: src/locales/en.json (hunk 7 above)
Comment (anuragajwani) [reply src/locales/en.json]: yes

Comment (stefaniacardenas) [reply src/locales/en.json]: So which command do you run to update the it? Should we add it to it to the package.json?

File: src/locales/en.json (hunk 1 above)
Comment (kopijunkie) [reply src/locales/en.json]: On the spreadsheet `webcam_permissions.allow` has been renamed to "Enable camera" (originally "Allow"), don't see it in use in the code as the button's label is currently mapped to `webcam_permissions.enable_webcam`. Probably should be using that other string map instead.

File: src/locales/en.json (hunk 7 above)
Comment (anuragajwani) [reply src/locales/en.json]: you just have to run `bisu` on the root of the project. It's a ruby gem. Is a package.json command something we want? If so how do we ensure in this project that ruby gem is installed prior to running the command ? I can't see a `Gemfile` in the project

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-812-pr-onfido-sdk-ui-onfido.surge.sh

File: src/locales/en.json (hunk 7 above)
Comment (anuragajwani) [reply src/locales/en.json]: on mobiles we only run it explicitly when required, i.e. string update. On the iOS SDK we have a Gemfile, on Android we don't. I assume on Android its a manual installation.

File: src/locales/en.json (hunk 1 above)
Comment (anuragajwani) [reply src/locales/en.json]: fixed

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-812-pr-onfido-sdk-ui-onfido.surge.sh

File: src/locales/en.json (hunk 2 above)
Comment (anuragajwani) [reply src/locales/en.json]: I switched them around on the mapping (not on the spreadsheet)

File: src/locales/en.json (hunk 3 above)
Comment (anuragajwani) [reply src/locales/en.json]: switched them around for the time being

File: src/locales/en.json (hunk 4 above)
Comment (anuragajwani) [reply src/locales/en.json]: fixed

**Comment by rfreitas** [comment]
Travis automatic deployment: https://812-pr-onfido-sdk-ui-onfido.surge.sh

File: src/locales/en.json [hunk 8]
```diff
@@ -382,16 +373,16 @@
...
   "short_passport": "passport",
   "short_driving_licence": "license",
   "short_national_identity_card": "card",
-  "bank_building_society_statement": "Bank/Building Society Statement",
+  "bank_building_society_statement": "Bank or building society statement",
   "utility_bill": "Utility Bill",
   "benefit_letters": "Benefits Letter",
   "government_letter": "Government Letter",
-  "council_tax": "Council Tax Letter",
+  "council_tax": "Council tax letter",
```
Comment (stefaniacardenas) [inline src/locales/en.json]: This title is inconsistent with the other ones that have capitalisation. Could you update the spreadsheet with `Council Tax Letter` please?

File: src/locales/en.json [hunk 9]
```diff
@@ -324,33 +315,33 @@
       "instruction": "All details should be clear and readable"
     },
     "sms_failed": {
-      "message": "Something's gone wrong",
-      "instruction": "Copy the below link to your mobile instead"
+      "message": "Something’s gone wrong",
```
Comment (stefaniacardenas) [inline src/locales/en.json]: Could you update the spreadsheet with a consistent apostrophe please? This one => `'`

Comment (anuragajwani) [reply src/locales/en.json]: fixed

File: src/locales/en.json (hunk 8 above)
Comment (anuragajwani) [reply src/locales/en.json]: fixed

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-812-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-812-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-812-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-812-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-812-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-812-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://812-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-812-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-812-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://812-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-812-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-812-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://812-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**